js/modules/k6/experimental/fs/module.go

```diff
@@ -0,0 +1,168 @@
+// Package fs provides a k6 module that allows users to interact with files from the
+// local filesystem.
+package fs
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/dop251/goja"
+	"go.k6.io/k6/js/common"
+	"go.k6.io/k6/js/modules"
+	"go.k6.io/k6/js/promises"
+	"go.k6.io/k6/lib/fsext"
+)
+
+type (
+	// RootModule is the global module instance that will create instances of our
+	// module for each VU.
+	RootModule struct {
+		registry *registry
+	}
+
+	// ModuleInstance represents an instance of the fs module for a single VU.
+	ModuleInstance struct {
+		vu modules.VU
+
+		registry *registry
+	}
+)
+
+var (
+	_ modules.Module   = &RootModule{}
+	_ modules.Instance = &ModuleInstance{}
+)
+
+// New returns a pointer to a new [RootModule] instance.
+func New() *RootModule {
+	return &RootModule{
+		registry: &registry{},
+	}
+}
+
+// NewModuleInstance implements the modules.Module interface and returns a new
+// instance of our module for the given VU.
+func (rm *RootModule) NewModuleInstance(vu modules.VU) modules.Instance {
+	return &ModuleInstance{vu: vu, registry: rm.registry}
+}
+
+// Exports implements the modules.Module interface and returns the exports of
+// our module.
+func (mi *ModuleInstance) Exports() modules.Exports {
+	return modules.Exports{
+		Named: map[string]any{
+			"open": mi.Open,
+		},
+	}
+}
+
+// Open opens a file and returns a promise that will resolve to a [File] instance
+func (mi *ModuleInstance) Open(path goja.Value) *goja.Promise {
+	promise, resolve, reject := promises.New(mi.vu)
+
+	// Files can only be opened in the init context.
+	if mi.vu.State() != nil {
+		reject(newFsError(ForbiddenError, "open() failed; reason: opening a file in the VU context is forbidden"))
+		return promise
+	}
+
+	if common.IsNullish(path) {
+		reject(newFsError(TypeError, "open() failed; reason: path cannot be null or undefined"))
+		return promise
+	}
+
+	// Obtain the underlying path string from the JS value.
+	pathStr := path.String()
+	if pathStr == "" {
+		reject(newFsError(TypeError, "open() failed; reason: path cannot be empty"))
+		return promise
+	}
+
+	go func() {
+		file, err := mi.openImpl(pathStr)
+		if err != nil {
+			reject(err)
+			return
+		}
+
+		resolve(file)
+	}()
+
+	return promise
+}
+
+func (mi *ModuleInstance) openImpl(path string) (*File, error) {
+	initEnv := mi.vu.InitEnv()
+
+	path = fsext.Abs(initEnv.CWD.Path, path)
+
+	fs, ok := initEnv.FileSystems["file"]
+	if !ok {
+		common.Throw(mi.vu.Runtime(), errors.New("open() failed; reason: unable to access the file system"))
+	}
+
+	if exists, err := fsext.Exists(fs, path); err != nil {
+		return nil, fmt.Errorf("open() failed, unable to verify if %q exists; reason: %w", path, err)
```

Comment:
The reasoning was that this was an "internal" error, as opposed to an error likely caused by the user's use of the module. 

In the line you referenced, we couldn't verify the file exists because some code internal to k6 failed to perform the operation. Whereas in the other branch, we were successful in checking the existence of the file, in which case we indicate to the user that the fs couldn't find the file.

Let me know if you have another solution/approach to this in mind 👍🏻